backend/sftp/sftp.go

```diff
@@ -788,37 +816,84 @@ func NewFsWithConnection(ctx context.Context, f *Fs, name string, root string, m
 	if err != nil {
 		return nil, fmt.Errorf("NewFs: %w", err)
 	}
-	cwd, err := c.sftpClient.Getwd()
-	f.putSftpConnection(&c, nil)
-	if err != nil {
-		fs.Debugf(f, "Failed to read current directory - using relative paths: %v", err)
-	} else if !path.IsAbs(f.root) {
-		f.absRoot = path.Join(cwd, f.root)
-		fs.Debugf(f, "Using absolute root directory %q", f.absRoot)
+	// Check remote shell type, try to auto-detect if not configured and save to config for later
+	if f.opt.ShellType != "" {
+		f.shellType = f.opt.ShellType
+		fs.Debugf(f, "Shell type %q from config", f.shellType)
+	} else {
```

Comment:
This means that all sftp backends will make an additional ssh command on startup. Often sftp is run without a config file so it won't be able to save the config.

I'm not sure this is an actual problem just thought I'd mention it!

Reply:
That is a fair point!

To avoid that I guess we would have to change it to "opt-in". Either default no shell access, but that would be a breaking change, so for backwards compat reasons maybe would have to assume unix shell by default. And then user could explicitely specify other shell types. And then this auto-detect logic could either be removed or moved to a separate option such as `--auto-detect-shell-or-something-like-that`.... Problem is when running the hash command, which rely on shell access, it would maybe fail with strange error messages before user realises he needs to configure different shell type.

Another alternative could be to try some sort of "delay-loading" this auto-detection part. That is, not run it on "startup", but first time it is needed - probably from the Hash command.

Not sure how these would work out, needs more thinking and experimenting...

Reply:
Oh, and the third alternative (or alternative zero), the current implementation, already have an "opt-out" switch, in that user can explicit set "shell_type", e.g. when running without config file as you say. Then the extra shell command will not be executed on startup. With an extra note in the docs, this could be considered an acceptable solution.

Reply:
I think a few docs to address this is fine.